docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.install

```diff
@@ -246,3 +246,92 @@ function hs_paragraph_types_update_10006() {
+    $field_config_storage = \Drupal::service('entity_type.manager')->getStorage('field_config');
+
+    /** @var \Drupal\Core\Entity\EntityStorageInterface $fieldConfigStorage */
+    $field_storage = \Drupal::service('entity_type.manager')->getStorage('field_storage_config');
+
+    // If storage does not yet exist, create it.
+    $storage_to_create = [
+      'paragraph.field_bg_color' => 'field.storage.paragraph.field_bg_color',
+      'paragraph.field_bg_color_width' => 'field.storage.paragraph.field_bg_color_width',
+    ];
+
+    foreach ($storage_to_create as $key => $fields) {
+      if (empty($field_storage->load($key))) {
+        $field_storage->createFromStorageRecord($config_storage->read($fields))->save();
+      }
+    }
+
+    $field_instances = [
+      'paragraph.hs_collection.field_bg_color' => 'field.field.paragraph.hs_collection.field_bg_color',
+      'paragraph.hs_collection.field_bg_color_width' => 'field.field.paragraph.hs_collection.field_bg_color_width',
+    ];
+
+    // Create the field instance.
+    foreach ($field_instances as $key => $instance) {
+      $config = $field_config_storage->load($key);
+      if (empty($config)) {
+        $field_config_storage->create($config_storage->read($instance))->save();
+      }
+    }
+
+    $displays = [
+      'core.entity_view_display.paragraph.hs_collection.default',
+      'core.entity_form_display.paragraph.hs_collection.default'
+    ];
+
+    foreach ($displays as $display) {
+      // Update displays.
+      $config = $config_factory->getEditable($display);
+      $new_config_data = $config_storage->read($display);
+      $config->setData($new_config_data);
+      $config->save();
+    }
+
+    // Get list of collection paragraphs.
+    $sandbox['ids'] = $paragraph_storage->getQuery()
+      ->accessCheck(FALSE)
+      ->condition('type', 'hs_collection')
+      ->execute();
+    $sandbox['total'] = count($sandbox['ids']);
+  }
+
+  $paragraph_ids = array_splice($sandbox['ids'], 0, 10);
+
+  // Set new field default for existing hs_collection paragraphs.
+  foreach ($paragraph_storage->loadMultiple($paragraph_ids) as $paragraph) {
+    /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+    if (
+      $paragraph->hasField('field_paragraph_style') &&
+      $paragraph->hasField('field_bg_color') &&
+      $paragraph->hasField('field_bg_color_width')) {
+      // Only set new field if old field had data.
+      if ($paragraph->get('field_paragraph_style')->value) {
+        $paragraph->set('field_bg_color', 'color_palette_default');
+      }
+      if ($paragraph->get('field_paragraph_style')->value == 'well-full-width') {
+        $paragraph->set('field_bg_color_width', 'full_width');
+      }
+      else {
+        $paragraph->set('field_bg_color_width', 'limited_width');
+      }
+
+      $paragraph->save();
+    }
+  }
+
+  $sandbox['#finished'] = count($sandbox['ids']) ? 1 - count($sandbox['ids']) / $sandbox['total'] : 1;
```

Comment:
You might be punished for all that's going on in this line.  You might get a better score from `if ($sandbox['ids']) {`...